docs/guides/real-time/chat/react.md

```diff
@@ -0,0 +1,419 @@
+
+```js
+const [formValue, setFormValue] = useState({ email: '', password: '' });
+```
+
+Set up a `handleLoginChange` method that updates the value of the login input field as the user types.
+
+```js
+const handleLoginChange = (event) => {
+	setFormValue({ ...formValue, [event.target.name]: event.target.value });
+};
+```
+
+Then, connect these values to the form input fields:
+
+```js
+<form onSubmit={loginSubmit}>
+	<label htmlFor="email">Email</label>
+	<input type="email" id="email" /> // [!code --]
+	<input type="email" id="email" name="email" value={formValue.email} onChange={handleLoginChange} /> // [!code ++]
+  
+	<label htmlFor="password">Password</label>
+	<input type="password" id="password" /> // [!code --] // [!code ++]
+	<input type="password" id="password" name="password" value={formValue.password} onChange={handleLoginChange} /> // [!code
+	++]
+	<button type="submit">Submit</button>
+</form>
+```
+
+Within the `loginSubmit` method, create a new WebSocket, which will immediately attempt connection:
+
+```js
+const loginSubmit = (event) => {
+	connectionRef.current = new WebSocket(url); // [!code ++]
+};
+```
+
+On connection, you must [send an authentication message before the timeout](/guides/real-time/authentication). Add an
+event handler for the connection's `open` event:
+
+```js
+const loginSubmit = (event) => {
+	connectionRef.current = new WebSocket(url);
+	connectionRef.current.onopen = () => authenticate(formValue); // [!code ++]
+};
+```
+
+Then, create a new `authenticate` method:
+
+```js
+const authenticate = (opts) => {
+	const { email, password } = opts;
+	connectionRef.current.send(JSON.stringify({ type: 'auth', email, password }));
+};
+```
+
+### Subscribe to messages
+
+In a WebSocket connection, all data sent from the server will trigger the connection’s `message` event. Inside
+`loginSubmit`, add an event handler:
+
+```js
+const loginSubmit = (event) => {
+	connectionRef.current = new WebSocket(url);
+	connectionRef.current.onopen = () => authenticate(formValue);
+	connectionRef.current.onmessage = (message) => receiveMessage(message); // [!code ++]
+};
+```
+
+Then, create a new `receiveMessage` method:
+
+```js
+const receiveMessage = (message) => {
+	const data = JSON.parse(message.data);
+};
+```
+
+As soon as you have successfully authenticated, a message will be sent. When this happens, subscribe to updates on the
+`Messages` collection. Add this inside of the `receiveMessage` method:
+
+```js {3-14}
+const receiveMessage = (message) => {
+	const data = JSON.parse(message.data);
+	if (data.type === 'auth' && data.status === 'ok') {
+		connectionRef.current.send(
+			JSON.stringify({
+				type: 'subscribe',
+				collection: 'messages',
+				query: {
+					fields: ['*', 'user_created.first_name'],
+					sort: 'date_created',
+				},
+			})
+		);
+	}
+};
+```
+
+When a subscription is started, a message will be sent to confirm. Add this inside of the `receiveMessage` method:
+
+```js {15-17}
```

Comment:
Same question here about highlighting type. This is only three lines, so feels like we should be calling it out more.